docs/guide/advanced-concepts/performance.md

```diff
@@ -0,0 +1,195 @@
+(xref-performance)=
+# Performance
+
+In this section, we will cover some of the important considerations when tuning
+your Bytewax dataflow for performance.
+
+Performance is a tricky subject to cover generally, as the goals for performance can
+vary across different use cases. As an example, some use-cases may prioritize latency,
+at the cost of overall throughput. In this guide, we will use the word performance to
+represent a desired goal, whether that goal is throughput or latency.
+
+As with all performance tuning, it is important to have good tooling to measure the
+effect of any changes you make. See the section on <project:#xref-profiling>
+for some tips on how to profile Bytewax.
+
+One of the worthwhile trade-offs of dataflow programming is having to play
+by the strict rules of operators and state management. In return you gain
+the ability to more-easily scale your dataflow across a cluster for increased
+performance. Unfortunately "more-easily" does not mean "trivially": Practically
+speaking, only adding more workers to a dataflow does not automatically result
+in an improvement in performance. The reality of all performance optimization
+is matching computational resources to the specific needs of your computation,
+and in this document we'll describe concrete design recommendations you should
+follow while implementing your dataflow to play to the strengths of Bytewax and
+ensure your dataflow is performant.
+
+## Bytewax's worker execution model
+
+A Bytewax **worker** is a thread that is executing your dataflow. Workers can be grouped
+into separate processes, but refer to a single thread within a single process.
+
+Bytewax's execution model uses identical workers. Workers execute all of the steps
+in a dataflow, including input and output steps. Workers **may** exchange
+data to ensure the semantics of operators, but do not automatically
+exchange work to other workers in order to increase performance.
+
+## Workers, processes and GIL contention
+
+When multiple workers are started within the same Python process using
+`-w/--workers-per-process`, they must contend over Python's GIL, or
+{external+python:std:term}`global interpreter lock`. Python's GIL ensures
+that only one Python thread can run at a time, preventing race conditions and
+ensuring thread-safety.
+
+In practice this means that when you are attempting to increase the parallelism
+of a dataflow, should try starting with multiple processes using the
+`-i/--process-id`, `-a/--addresses` instead of multiple workers with `-w`.
```

Comment:
```suggestion
In practice this means that when you are attempting to increase the parallelism
of a dataflow, you should try starting with multiple processes using the
`-i/--process-id`, `-a/--addresses` instead of multiple workers with `-w`.
```